tutorials/source-modeling/plot_beamformer_lcmv.py

```diff
@@ -0,0 +1,169 @@
+
+# sphinx_gallery_thumbnail_number = 5
+
+import mne
+from mne.datasets import sample
+from mne.beamformer import make_lcmv, apply_lcmv
+
+###############################################################################
+# Data processing
+# ---------------
+# We will use the sample data set for this tutorial and reconstruct source
+# activity on the trials with left auditory stimulation. Note that
+# beamformers are usually computed in a :class:`volume source space
+# <mne.VolSourceEsitmate>`, as a visualization
+# of only the surface activation can misrepresent the data.
+
+data_path = sample.data_path()
+subjects_dir = data_path + '/subjects'
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+fwd_fname = data_path + '/MEG/sample/sample_audvis-meg-vol-7-fwd.fif'
+
+# Read forward model
+forward = mne.read_forward_solution(fwd_fname)
+
+# Read the raw data
+raw = mne.io.read_raw_fif(raw_fname, preload=True)
+raw.info['bads'] = ['MEG 2443']  # bad MEG channel
+
+# Set up the epoching
+event_id = 1  # those are the trials with left-ear auditory stimuli
+tmin, tmax = -0.2, 0.5
+events = mne.find_events(raw)
+
+# pick relevant channels
+raw.pick(['meg', 'eog'])  # pick channels of interest
+
+# Create epochs
+proj = False  # already applied
+epochs = mne.Epochs(raw, events, event_id, tmin, tmax,
+                    baseline=(None, 0), preload=True, proj=proj,
+                    reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6))
+
+# Visualize averaged sensor space data
+evoked = epochs.average()
+evoked.plot_joint()
+
+###############################################################################
+# Computation of covariance matrices
+# ----------------------------------
+# The spatial filter is computed from two ingredients: the forward model
+# solution that we read from disk above and the covariance matrix of the data.
+# The data covariance matrix will be `inverted`_ during the spatial filter
+# computation, so it is valuable to plot the covariance matrix and its
+# eigenvalues to gauge whether matrix inversion will be possible.
+# Also, because we want to combine different channel types (magnetometers and
+# gradiometers), we need to account for the different amplitude scales of these
+# channel types. To do this we will supply a noise covariance matrix to the
+# beamformer, which will be used for whitening.
+
+data_cov = mne.compute_covariance(epochs, tmin=0.05, tmax=0.25,
+                                  method='empirical')
+noise_cov = mne.compute_covariance(epochs, tmin=tmin, tmax=0,
+                                   method='empirical')
+
+data_cov.plot(epochs.info)
+
+###############################################################################
+# Compute the spatial filter
+# --------------------------
+# Now we can compute the spatial filter.
+# When looking at the covariance matrix plots, we can see that our data is
+# slightly rank-deficient. Thus, we will regularize the covariance matrix by
```

Comment:
I would include this information right after the covariance plots (before beginning the next subsection).  Would be great to also say what exactly it is about the plots that indicates rank-deficiency.

This might be overkill for this tutorial, but plotting the covariance matrix *after* the 5% regularization would be really informative for many users I think.

Reply:
Ah I will have to move this, overlooked before.

Reply:
(Done now.)